### PR TITLE
Add tests for getConfig

### PR DESCRIPTION
**Pull request by jasonlafferty**
## Description

Add unit tests around `getConfig`, this includes refactoring to be able to test better. 

## Changes/Tasks

- [x] Add tests around `getConfig`

## Motivation and Context

This is the beginning of epic #439 which is to add more testing around `react-static`.

## Types of changes

- [x] Refactoring/add tests (refactoring or adding test which isn't a fix or add a feature)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [x] My changes have tests around them

### CODE REVIEW

File: src/static/__tests__/getConfig.test.js [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      expect(routes).toEqual([
+        { hasGetProps: false, noindex: undefined, path: '/' },
+        { is404: true, path: '404' },
+      ])
+    })
+  })
+})
+
+describe('buildConfigation', () => {
+  it('production', () => {
```
Comment (tannerlinsley) [inline src/static/__tests__/getConfig.test.js]: Should this be development?

File: src/static/__tests__/getConfig.test.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+import defaultConfigDevelopment from '../__mocks__/defaultConfigDevelopment.mock'
+import defaultConfigProduction from '../__mocks__/defaultConfigProduction.mock'
+
+jest.mock('path', () => ({
+  resolve: (stringOne = '/', stringTwo = '') => `${stringOne}${stringTwo}`,
+  join: (stringOne, stringTwo) => `${stringOne}/${stringTwo}`,
+  dirname: () => 'root/',
+}))
+
+jest.mock('../../utils/getDirname', () => () => './dirname/')
```
Comment (jasonlafferty) [inline src/static/__tests__/getConfig.test.js]: not too sure about this, anyone who knows a better way to mock `__dirname`?

**Comment by jasonlafferty** [comment]
@tannerlinsley unsure about deployment being broken. Have written some test for `getConfig` and refactored some, could you have a look, cheers. Thanks.

**Comment by tannerlinsley** [comment]
This looks fantastic. Ignore the deploy preview. Gitdocs is broken again 😔. I'm good to merge if you are!

**Comment by jasonlafferty** [comment]
should be fine to merge if you've checked behaviour is the same 👍.

**Comment by tannerlinsley** [comment]
Needs to be fixed up

**Comment by tannerlinsley** [comment]
What's the status on this @jasonlafferty ?

**Comment by jasonlafferty** [comment]
@tannerlinsley, they got out of date. had not time. Will try update them this week and will message.

**Pull request merged**